OpenRA.Mods.Common/Widgets/Logic/Ingame/IngameChatLogic.cs

```diff
@@ -71,7 +72,46 @@ public IngameChatLogic(Widget widget, OrderManager orderManager, World world, Mo
 			var chatMode = chatChrome.Get<ButtonWidget>("CHAT_MODE");
 			chatMode.GetText = () => teamChat ? "Team" : "All";
 			chatMode.OnClick = () => teamChat ^= true;
-			chatMode.IsDisabled = () => disableTeamChat;
+
+			// Team chat is disabled if we are the only spectator
+			// This changes as soon as a defeated player can talk in the spectator chat
+			if (!alwaysDisabled && isOnlyObserver)
+			{
+				chatMode.IsDisabled = () =>
+				{
+					if (world.IsGameOver)
+						return true;
+
+					disableTeamChat = players.All(p => p.WinState == WinState.Undefined);
+					return disableTeamChat;
+				};
+			}
+			else if (!alwaysDisabled && world.LocalPlayer != null)
+			{
+				chatMode.IsDisabled = () =>
+				{
+					if (world.IsGameOver)
+						return true;
+
+					// Nothing changed since the start
+					if (world.LocalPlayer.WinState == WinState.Undefined)
+						return disableTeamChat;
+
+					// At this point our player is dead
+					// Allow to chat with existing spectators
+					if (observersExist)
+					{
+						disableTeamChat = false;
+						return disableTeamChat;
+					}
+
+					// Or wait until another player died
+					disableTeamChat = players.All(p => p.WinState == WinState.Undefined);
```

Comment:
This is guaranteed to return false because the local player has lost. Shouldn't this also be requiring `p != world.LocalPlayer`?

Reply:
`players` does not contain the local player.